### PR TITLE
[Ubuntu] Add Ubuntu-24.04 base image

### PR DESCRIPTION
**Pull request by erik-bershel**
# Description

Due to release of the new Ubuntu Noble Numbat we are going to release related runner image.

#### Related issue:

https://github.com/actions/runner-images/issues/9691

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Comment by HonkingGoose** [comment]
I apologize in advance, as this may be a silly question. The [`readme.md` does _not_ list the Ubuntu 24.04 as an available image](https://github.com/actions/runner-images?tab=readme-ov-file#available-images). Is that what you really wanted?

I kinda expected this PR to put `ubuntu-24.04` in the "Available Images" section of the readme. But maybe there is more work that needs to be done before this image can be released as a beta to the general public?

**Comment by dmytrosklyarov** [comment]
Hi, am I the only one experiencing issues with Ubuntu 22.04 image generation with the next error after the merge of this PR: 

`==> azure-arm.build_image: /imagegeneration/SoftwareReport/Generate-SoftwareReport.ps1 : The term 'erl' is not recognized as a name of a cmdlet, function, script file, or executable program.
==> azure-arm.build_image: Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
==> azure-arm.build_image: + CategoryInfo          : ObjectNotFound: (erl:String) [Generate-SoftwareReport.ps1], CommandNotFoundException
==> azure-arm.build_image: + FullyQualifiedErrorId : CommandNotFoundException,Generate-SoftwareReport.ps1
`

File: images/ubuntu/scripts/tests/Tools.Tests.ps1 [hunk 1]
```diff
@@ -144,7 +145,7 @@ Describe "Cmake" {
     }
 }
 
-Describe "erlang" -Skip:(Test-IsUbuntu22) {
+Describe "erlang" -Skip:(-not (Test-IsUbuntu20)) {
```
Comment (Thlbi) [inline images/ubuntu/scripts/tests/Tools.Tests.ps1:148]: Hello,
This test is not skipped in Ubuntu22 causing the following problem: `
==> azure-arm.build_image: /imagegeneration/SoftwareReport/Generate-SoftwareReport.ps1 : The term 'erl' is not recognized as a name of a cmdlet, function, script file, or executable program. ==> azure-arm.build_image: Check the spelling of the name, or if a path was included, verify that the path is correct and try again. ==> azure-arm.build_image: + CategoryInfo          : ObjectNotFound: (erl:String) [Generate-SoftwareReport.ps1], CommandNotFoundException ==> azure-arm.build_image: + FullyQualifiedErrorId : CommandNotFoundException,Generate-SoftwareReport.ps1`

**Comment by joaopedrocg27** [comment]
Hey @erik-bershel 

Why was apt-fast removed from ubuntu 24? Is this now provided by default or there were other concerns?

**Comment by andyfeller** [comment]
@mikhailkoliada @erik-bershel : could either of you provide more context why a number of previously available apt packages on ubuntu-22.04 were not carried over on ubuntu-24.04?

**Comment by erik-bershel** [comment]
@andyfeller more info about reasons for changes we provide on the release:
- https://github.com/actions/runner-images/issues/9848

**Comment by andyfeller** [comment]
Thank you so much, @erik-bershel!  All of those reasons make sense.  I primarily ask as an infrequent workflow I was trying to use broke because `imagemagick` wasn't present.  While I figured if its use case really justified trying to request adding it back, I just wanted to understand for future purposes.

**Pull request merged**